presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DeterminismAnalyzer.java

```diff
@@ -90,14 +91,17 @@ public DeterminismAnalyzer(
         this.handleLimitQuery = config.isHandleLimitQuery();
     }
 
-    protected DeterminismAnalysisDetails analyze(QueryObjectBundle control, ChecksumResult controlChecksum)
+    protected DeterminismAnalysisDetails analyze(QueryObjectBundle control, ChecksumResult controlChecksum, QueryAction overrideActionForQuery)
     {
+        // In case the action is not specified, use the default one we were constructed with.
```

Comment:
i think it would make sense to remove the queryAction field from the class and then always pass in the queryAction here (never let it be null).

Reply:
@rschlussel 
We will still need HelperAction to run checksum and other types of helper queries.
I wonder if we can refactor this in the followup PR?
This PR would really help us unload prod clusters sooner and not break any existing use cases.

Reply:
can you do it in a second commit in this PR? That way revert is easier if something breaks with the refactoring, but if nothing breaks, we leave things in a better state.

Reply:
@rschlussel 
Actually, we might want to leave the `queryAction` field, but rename it to the `helperAction`, because we want to use it to run checksums regardless of using control or test for determinism analysis.
But we enforce `overrideActionForQuery` (rename it to `queryAction`) to be non-null and use it for issuing the query itself, like you are proposing.

Sounds good?
I'll try adding most of this in the second commit.